Form.php

```diff
@@ -0,0 +1,357 @@
+   *
+   * @return string
+   *  Sanitized string '$data'.
+   */
+  public static function testInput(string $data): string {
+    $data = trim($data);
+    $data = stripslashes($data);
+    return htmlspecialchars($data);
+  }
+
+  /**
+   * Method to validate 'Subject|Marks' pairs.
+   *
+   * @param string $data
+   *  Data to be verified.
+   * @param string $regex
+   *  Regex to be validated against.
+   *
+   * @return string
+   *  Stating error message(if any).
+   */
+  function checkSubjectMarks($data, $regex) {
+    // No data is entered.
+    if (empty($data)) {
+      return "No subject|marks pairs entered";
+    }
+    // If input contains forbidden characters.
+    elseif (!preg_match($regex, $data)) {
+      return "Only alphabets, digits and | allowed";
+    }
+    // Validating each Subject|Marks pair.
+    else {
+      $lines = explode("\n", $data);
+      $numberOfLines = count($lines);
+      for ($i = 0; $i < $numberOfLines; $i++) {
+        $subjectMarksPair = explode("|", $lines[$i]);
+        // Trimming amy extra space in subject and marks.
+        $subject = trim($subjectMarksPair[0], " \t\n\r\0\x0B");
+        $marks = trim($subjectMarksPair[1], " \t\n\r\0\x0B");
+        // If Subject is numeric or Marks is non-numeric or incorrect format.
+        if (count($subjectMarksPair) !== 2 || is_numeric($subject) || !is_numeric($marks)) {
+          return "Invalid Pairs";
+        }
+        // If marks greater than 100 is entered.
+        elseif ($subjectMarksPair[1] > 100) {
+          return "Marks cannot be greater than 100";
+        }
+      }
+    }
+    // No error.
+    return "";
+  }
+
+  /**
+   * Method to validate Indian Phone Number.
+   *
+   * @param string $phoneNumber
+   *  Phone number to be validated.
+   * @param string $phoneNumberRegex
+   *  Regex to be validated against.
+   *
+   * @return string
+   *  Stating error message(if any).
+   */
+  function indianPhoneCheck($phoneNumber, $phoneNumberRegex): string {
+    $phoneNumber = trim($phoneNumber, " \t\n\r\0\x0B");
+    // No user input.
+    if(empty($phoneNumber)) {
+      return "Phone number is required";
+    }
+    // Contains invalid character(s).
+    else if (!preg_match($phoneNumberRegex, $phoneNumber)) {
+      return "Only digits and + symbol allowed";
+    }
+    // No error.
+    return "";
+  }
+
+  /**
+   * Method to validate email address.
+   *
+   * @param string $email
+   *  Email address to be validated.
+   *
+   * @return string
+   *  Stating error message(if any).
+   */
+  function checkEmailAddress (string $email): string {
+    // Trimming extra spaces.
+    $email = trim($email, " \t\n\r\0\x0B");
+    // No user input.
+    if(empty($email)) {
+      return "Email address is required";
+    }
+    // Invalid email syntax.
+    else if (filter_var($email, FILTER_VALIDATE_EMAIL) == false) {
+      return "Email address is invalid";
+    }
+    else {
+      // Initialising curl.
+      $ch = curl_init();
```

Comment:
Why don't  we using guzzlehttp?

Reply:
I wanted to explore curl as an alternative to guzzlehttp/guzzle.